devlib/target.py

```diff
@@ -2991,9 +2993,11 @@ def __init__(self,
         # Pull out adb connection settings
         adb_conn_params = ['device', 'adb_server', 'timeout']
         self.android_connection_settings = {}
-        for setting in adb_conn_params:
-            if connection_settings.get(setting, None):
-                self.android_connection_settings[setting] = connection_settings[setting]
+        self.android_connection_settings.update(
```

Comment:
So we established we want to include the values that are falsy in `android_connection_settings` ? That sounds reasonable but we will need to go through the list of keys and check that for each of them a falsy value is valid and not signalling anything special. For example, we might have a higher layer that sets a bunch of None that would be expected to be filtered at this stage, or maybe not.

Reply:
Question answered above and the result is yes we want to propagate falsy value, so the comprehension is appropriate